test/integration/stats_integration_test.cc

```diff
@@ -195,8 +195,22 @@ TEST_P(ClusterMemoryTestRunner, MemoryLargeClusterSizeWithStats) {
 
   EXPECT_LT(start_mem, m1);
   EXPECT_LT(start_mem, m1001);
-  // As of 2019/04/12, m_per_cluster = 59576 (libstdc++)
-  EXPECT_LT(m_per_cluster, 59600);
+
+  // Note: if you are increasing this golden value because you are adding a
+  // stat, please confirm that this will be generally useful to most Envoy
+  // users. Otherwise you are adding to the per-cluster memory overhead, which
+  // will be significant for Envoy installations that are massively
+  // multi-tenant.
+  //
+  // History of golden values:
+  //
+  // Date        PR       Bytes Per Cluster   Notes
+  // ----------  -----    -----------------   -----
+  // 2019/03/20  6329     59015               Initial version
+  // 2019/04/12  6477     59576               Implementing Endpoint lease...
+  // 2019/04/23  6659     59512               Reintroduce dispatcher stats...
+
+  EXPECT_EQ(m_per_cluster, 59512);
```

Comment:
Do we only run this on one particular build type and compiler? Just wondering how we can be so precise. Looks good otherwise.

Reply:
Well the compiler doesn't appear to matter than much, at least between g++ and clang. We do run only in one STL impl -- at one point in @cmluciano 's PR we were trying to work with two different libraries but abandoned that for reasons I forgot.

If any stability issues arise on any particular platform, we should adjust hasDeterministicMallocStats to skip that one.

Reply:
Don't we use `libstdc++` on OS X?

Reply:
Maybe we use that, but then either: hasDeterministicMallocStats() returns false for some other reason (e.g. not using TCMALLOC) or libc++ has exactly the same memory usage for the operations this is measuring, as CI seems to work on Mac.

Reply:
I'll merge, but I think this whole situation is very fragile and will probably break new architectures/compilers. I hope we aren't using `hasDeterministicMallocStats()` for any compiler or STL with a different layout than the one we expect.

Reply:
Thanks Harvey -- I think it will be a pretty clear test failure if that happens that could be addressed by changing the definition of hasDeterministicMallocStats() (and maybe its name...maybe 'deterministic' is the wrong name for that....)

In the meantime, I've noticed since starting on this journey of trying to prune memory that it's bloated quite a bit and so I'm hoping this makes us more conscious of costs.